### ISSUE TITLE
orthogonal views

### ISSUE THREAD

**Issue by edljk** [JuliaPlots/Makie.jl#156]
Is it possible to switch from a 3D plot to a 2D view. `axis2d!` plays this kind of role but is it possible to choose another orthogonal (orthographic) projection: for instance a view of the projection on `(0xz)` instead of `(0xy)`?

**Comment by edljk** [comment]
I tried something below but this does not seem to be really orthogonal / orthographic and I do not know how to adjust the height `1.8` to be optimal for the window.
Of course reflecting the axis labels would also be great! Any suggestion would be welcome..

```julia
x = Vec3f0(0); baselen = 0.2f0; dirlen = 1f0
# create an array of differently colored boxes in the direction of the 3 axes
rectangles = [
    (HyperRectangle(Vec3f0(x), Vec3f0(dirlen, baselen, baselen)), RGBAf0(1,0,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, dirlen, baselen)), RGBAf0(0,1,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, baselen, dirlen)), RGBAf0(0,0,1,1))
]
meshes = map(GLNormalMesh, rectangles)
scene = mesh(merge(meshes))
Makie.cam3d!(scene)
Makie.center!(scene)

cam = Makie.cameracontrols(scene)
dir = scene.limits[].widths ./ 2.
dir_scaled = Float32[dir[1] * scene.transformation.scale[][1],
                     dir[2] * scene.transformation.scale[][2],
                     0.]
cam.upvector[] = Float32[0., 1., 0.]
cam.lookat[] = scene.limits[].origin + dir_scaled
cam.eyeposition[] = Float32[cam.lookat[][1], cam.lookat[][2], cam.lookat[][3] - 1.8]
Makie.update_cam!(scene, cam)
```

**Comment by ffreyer** [comment]
There is a keyword argument `projectiontype` which takes `AbstractPlotting.Perspective` or `AbstractPlotting.Orthographic`. It seems like this hasn't been implemented fully yet though, since it still refers to GLAbstraction (which hasn't been updated to 0.7/1.0)

A quick workaround would be picking a small (non-zero) fov, e.g. `Makie.cam3d!(scene, fov = 1f0)`. This will make the projection nearly orthographic, but mess up the initial zoom level. It also causes some clipping issues for me in your example, for some reason.

**Comment by SimonDanisch** [comment]
@ffreyer fixed this :) 

You should be able to use this on master (AbstractPlotting & Makie):
```Julia
using GeometryTypes
x = Vec3f0(0); baselen = 0.2f0; dirlen = 1f0
# create an array of differently colored boxes in the direction of the 3 axes
rectangles = [
    (HyperRectangle(Vec3f0(x), Vec3f0(dirlen, baselen, baselen)), RGBAf0(1,0,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, dirlen, baselen)), RGBAf0(0,1,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, baselen, dirlen)), RGBAf0(0,0,1,1))
]
meshes = map(GLNormalMesh, rectangles)
scene = mesh(merge(meshes))
cam = Makie.cameracontrols(scene)
dir = scene.limits[].widths ./ 2.
dir_scaled = Float32[dir[1] * scene.transformation.scale[][1],
                     dir[2] * scene.transformation.scale[][2],
                     0.]
cam.upvector[] = Float32[0., 1., 0.]
cam.lookat[] = scene.limits[].origin + dir_scaled
cam.eyeposition[] = Float32[cam.lookat[][1], cam.lookat[][2], cam.lookat[][3] - 1.8]
cam.projectiontype[] = AbstractPlotting.Orthographic
scene
```

**Comment by edljk** [comment]
Great thanks @ffreyer! 
By the way, do you have an idea of how to optimize the `1.8` value to fit at best the projection of the original object?

**Comment by ffreyer** [comment]
No, sorry. I only use orthographic projections for one type of visualization, where the dimensions are almost always the same. So I just set some fitting defaults and zoom manually otherwise.

**Comment by edljk** [comment]
Yes, manual zooming could be an option but it is freezed after previous setting of the camera.. and produces the following error on the command line:
```julia
julia> Failed to push!
    (0.0, -1.0)
to node
    7: "scroll" = (0.0, 0.0) Tuple{Float64,Float64} (active)

error at node: 270: "map(scroll)" = nothing Nothing (active)
UndefVarError: GLAbstraction not defined
translate_cam!(::Scene, ::Camera3D, ::Vec{3,Float32}) 
...
```

**Comment by SimonDanisch** [comment]
Are you sure that you are on Makie & AbstractPlotting master? If I'm not mistaken, you run into the error @ffreyer fixed... I'm actually pretty sure that you aren't, since Makie still seems to use Reactive nodes (as can be seen by the printing) instead of Observables ;)

**Comment by ffreyer** [comment]
Yea that's exactly what the PR fixed. You can switch with 
```julia
pkg> add Makie#master
pkg> AbstractPlotting#master
```

**Comment by edljk** [comment]
Really sorry I was not on master..
No errors any more when zooming but I am surprised by the result: I expected a `y` view associated to `cam.upvector[] = Float32[0., 1., 0.]`..
![projaxis](https://user-images.githubusercontent.com/14992507/47417139-9c68e400-d777-11e8-8f88-40d07cf6e9a8.png)

**Comment by SimonDanisch** [comment]
I get:
![image](https://user-images.githubusercontent.com/1010467/47418287-1d28df80-d77a-11e8-8895-9acfa4c855b7.png)

```julia
using GeometryTypes
x = Vec3f0(0); baselen = 0.2f0; dirlen = 1f0
# create an array of differently colored boxes in the direction of the 3 axes
rectangles = [
    (HyperRectangle(Vec3f0(x), Vec3f0(dirlen, baselen, baselen)), RGBAf0(1,0,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, dirlen, baselen)), RGBAf0(0,1,0,1)),
    (HyperRectangle(Vec3f0(x), Vec3f0(baselen, baselen, dirlen)), RGBAf0(0,0,1,1))
]
meshes = map(GLNormalMesh, rectangles)
scene = mesh(merge(meshes))
display(scene)
cam = Makie.cameracontrols(scene)
dir = scene.limits[].widths ./ 2.
dir_scaled = Vec3f0(
    dir[1] * scene.transformation.scale[][1],
    dir[2] * scene.transformation.scale[][2],
    0.0
)
cam.upvector[] = (0.0, 1.0, 0.0)
cam.lookat[] = scene.limits[].origin + dir_scaled
cam.eyeposition[] = Vec3f0(cam.lookat[][1], cam.lookat[][2], cam.lookat[][3] - 1.8)
cam.projectiontype[] = AbstractPlotting.Orthographic
update_cam!(scene, cam)
```

**Comment by ffreyer** [comment]
Same for me, both on the latest release an master. Though I did get confused by the axis labeling for a bit. My intuition was that y would be large where the y - label is, which is not true here. Maybe @edljk got confused by the same thing?

Maybe an axis style like this (without the colors) would be more intuitive?

![image](https://user-images.githubusercontent.com/10947937/47421849-b7d8ec80-d781-11e8-9432-2d516cb33f77.png)

**Comment by edljk** [comment]
ok, I get now the same view as Simon (after manual dezooming), thanks. I do not know what happened. Will try to fix the `1.8` coefficient.
